src/clj/puppetlabs/trapperkeeper/services/protocols/filesystem_watch_service.clj

```diff
@@ -4,8 +4,11 @@
 
 (def Event
   "Schema for an event on a file watched by this service."
-  {:type (schema/enum :create :modify :delete :unknown)
-   :path File})
+  (schema/if #(= (:type %) :unknown)
+    {:type (schema/eq :unknown)
+     :path (schema/eq nil)}
```

Comment:
Yea, I can't find any examples, so maybe I made it up. `schema/pred` is more common than `schema/eq` in general, but `(= nil thing)` and `(nil? thing)` should be equivalent, so :shrug:.